### PR TITLE
[Model] Update the `MoDTransformerBlock` and add a small example.

### PR DESCRIPTION
**Pull request by keli-wen**
## Description

Align with MoD paper.

> Section 5. "In this work we show that one can successfully use a top-k routing scheme during training, but not require it during later autoregressive sampling. Either a simple auxiliary classifier, or auxiliary loss on the router, is sufficient to learn the top-𝑘 routing decisions such that it can mimic the top-𝑘 decisions during autoregressive sampling, with minimal to no performance degradation."

- [x] Use real top-k decision in the train stage.
- [x] Use auxiliary router prediction decision in the eval stage.

Add a minimal example in the main function scope.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the `lint.sh` to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] Code is well-documented.

### CODE REVIEW

File: MoD/MoD.py [hunk 1]
```diff
@@ -109,26 +113,54 @@ def forward(self, x: torch.Tensor):
...
+        topk_decision = (
+            aux_targets.bool() if self.training else aux_decision.bool()
+        )
 
+        # TODO[keli]: How to enable batch processing for the following loop?
         for b in range(B):
             # Extract tokens and router that need to go through the transformer block.
-            # [Shape] selected_tokens_emb: (selected_tokens_count, hidden_size)
-            selected_tokens_emb = x[b, aux_decision[b]]
+            # `unsqueeze(0)` is used to add the batch dimension back.
```
Comment (NANWOOD) [inline MoD/MoD.py:125]: Maybe unnecessary.

**Review by NANWOOD — APPROVED** [review]
The comment is optional, LGTM.

File: MoD/MoD.py (hunk 1 above)
Comment (keli-wen) [reply MoD/MoD.py:125]: Great, but I think the code is suck, we can update the whole code later.
